scripts/_compat.sh

```
@@ -42,8 +42,282 @@ has_inotify() {
+      continue
+    fi
+    sleep 0.2
+    waited=$(( waited + 1 ))
+    if [[ $waited -ge $(( wait_max * 5 )) ]]; then
+      echo "[_compat] WARN: lock timeout on $lockdir" >&2
+      return 1
+    fi
+  done
+  # shellcheck disable=SC2064
+  trap "rmdir '$lockdir' 2>/dev/null || true" RETURN
+  eval "$@"
```

Shellcheck (0.11.0)

[warning] eval negates the benefit of arrays. Drop eval to preserve whitespace/symbols (or eval as string).

(SC2294)